src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -486,6 +486,10 @@ Copyright (c) .NET Foundation. All rights reserved.
                                     Condition="'$(ServerGarbageCollection)' != ''"
                                     Value="$(ServerGarbageCollection)" />
 
+    <RuntimeHostConfigurationOption Include="System.GC.DynamicAdaptationMode"
+                                    Condition="'$(GarbageCollectionAdaptationMode)' != ''"
```

Comment:
we should determine a better name here, have made it similar to `ServerGarbageCollection` used above.

Reply:
What is our policy to determine which public configuration property gets exposed through MSBuild or not? We have some other properties that might make sense to expose here (e.g. GCConserveMem) while some other might not (e.g. RetainVM)

Reply:
yeah not quite sure about the history around this, given the naming differences I am guessing different teams ended up implementing this. We could inventory what needs to be added and happy to add them so we get some consistent naming here.

Reply:
since this kind of tasks were not historically done by the GC team, my guess is what used to be in the hosting startup flags were made available in the msbuild options which meant Concurrent GC, Server GC and RetainVM (which was an important thing in the 32-bit days but no longer relevant in 64-bit).

agreed that we should do an inventory but that shouldn't delay the checkin of this change. perhaps open an issue on the inventory part?

Reply:
yeah sounds good, I can create an issue to inventory settings. Once @mrsharm validates that the new runtimeconfig settings are working, I can merge this in. I have validated that the msbuild property adds the value withing runtimeconfig.